cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2245,7 +2245,7 @@ endif()
 if(ARROW_USE_XSIMD)
   resolve_dependency(xsimd
                      REQUIRED_VERSION
-                     "8.1.0"
+                     "9.0.0"
```

Comment:
@cyb70289 Should we drop support for xsimd 8.1.0? Or is it OK that we support xsimd 8.1.0 or later?

Reply:
I think we can drop support for xsimd 8.1.0. It's okay to only support the latest version.

Reply:
OK. Thanks for sharing your opinion.

If our xsimd 8.1.0 related maintenance cost isn't high, I want to keep xsimd 8.1.0 support because some packaging systems doesn't provide xsimd 9.0.0 yet. e.g.:

* Debian: https://packages.debian.org/search?keywords=xsimd
* Conan: https://conan.io/center/xsimd

Is xsimd 8.1.0 support troublesome for us? It seems that the current our code base can work with xsimd 8.1.0 and 9.0.0.

Reply:
Ah, then it's better to keep supporting xsimd 8.1.0. There's no trouble for us.

Reply:
Thanks!

Reply:
@serge-sans-paille Could you revert this change to support xsimd 8.1.0 too.

Reply:
Should be of now, if I  correctly understand the codebase.